azurelinuxagent/ga/exthandlers.py

```diff
@@ -60,7 +61,8 @@
 
 _VALID_HANDLER_STATUS = ['Ready', 'NotReady', "Installing", "Unresponsive"]
 
-HANDLER_NAME_PATTERN = re.compile(_HANDLER_PATTERN + r'$', re.IGNORECASE)
+HANDLER_NAME_PATTERN = re.compile(_HANDLER_NAME_PATTERN, re.IGNORECASE)
+HANDLER_COMPLETE_NAME_PATTERN = re.compile(_HANDLER_PATTERN + r'$', re.IGNORECASE)
```

Comment:
It does make more sense for this flag to be in extension_telemetry.py but I didnt put it there because that creates a circular dependency (because I need to read the HANDLER_NAME from exthandlers.py) and moving these globals (HANDLER_NAME, etc) outside doesn't make logical sense either since they're defined for the Extension Handlers. 

To avoid all this confusion I just left the  is_extension_telemetry_pipeline_enabled  in exthandlers because that's where we are reading the flag. If you feel strongly about it I can move it to a common place though (maybe AgentGlobals.py or version.py) to avoid the circular dependency.